Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -2359,6 +2359,167 @@ final class BuildPlanTests: XCTestCase {
+                Manifest.createFileSystemManifest(
+                    name: "zPkg",
+                    path: .init("/zPkg"),
+                    products: [
+                        ProductDescription(name: "Z", type: .library(.automatic), targets: ["Z"]),
+                    ],
+                    targets: [
+                        TargetDescription(name: "Z", dependencies: ["Foo"]),
+                        TargetDescription(name: "Foo", dependencies: []),
+                    ]),
+                Manifest.createFileSystemManifest(
+                    name: "yPkg",
+                    path: .init("/yPkg"),
+                    dependencies: [
+                        .localSourceControl(path: .init("/zPkg"), requirement: .upToNextMajor(from: "1.0.0")),
+                    ],
+                    products: [
+                        ProductDescription(name: "Y", type: .library(.automatic), targets: ["Y"]),
+                    ],
+                    targets: [
+                        TargetDescription(name: "Y",
+                                          dependencies: [
+                                            .product(name: "Z",
+                                                     package: "zPkg"
+                                          )]),
+                    ]),
+                Manifest.createFileSystemManifest(
+                    name: "xPkg",
+                    path: .init("/xPkg"),
+                    dependencies: [
+                        .localSourceControl(path: .init("/yPkg"), requirement: .upToNextMajor(from: "1.0.0")),
+                    ],
+                    products: [
+                        ProductDescription(name: "X", type: .library(.automatic), targets: ["X"]),
+                    ],
+                    targets: [
+                        TargetDescription(name: "X",
+                                          dependencies: [.product(name: "Y",
+                                                                  package: "yPkg"
+                                                                 )
+                                          ]),
+                    ]),
+                Manifest.createRootManifest(
+                    name: "appPkg",
+                    path: .init("/appPkg"),
+                    dependencies: [
+                        .localSourceControl(path: .init("/aPkg"), requirement: .upToNextMajor(from: "1.0.0")),
+                        .localSourceControl(path: .init("/xPkg"), requirement: .upToNextMajor(from: "1.0.0")),
+                    ],
+                    targets: [
+                        TargetDescription(name: "App",
+                                          dependencies: [ "Foo",
+                                                         .product(name: "A",
+                                                                  package: "aPkg",
+                                                                  moduleAliases: ["Foo": "FooA"]),
```

Comment:
the updated propagation method allows aliasing targets that are multi-level upstream at the user level